crypto/vrf/compatibility_keys_test.go

```diff
@@ -0,0 +1,210 @@
+//go:build libsodium
+// +build libsodium
+
+package vrf
+
+import (
+	"bytes"
+	"crypto/ed25519"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	coniks "github.com/coniks-sys/coniks-go/crypto/vrf"
+	ed25519consensus "github.com/hdevalence/ed25519consensus"
+	libsodium "github.com/line/ostracon/crypto/vrf/internal/vrf"
+	xed25519 "golang.org/x/crypto/ed25519"
+)
+
+var secret [SEEDBYTES]byte
+var message []byte = []byte("hello, world")
+
+func TestKeygen(t *testing.T) {
```

Comment:
I am curious and have a question. If `curve25519-voi` is used in the future, will that test be added as well?

Reply:
Yes, it does. I've already prepared that. But `curve25519-voi` needs `golang-1.16+`, so I've not put the commit these updates yet.

```diff
+       t.Run("ed25519 and voied25519 have compatibility",
+               func(t *testing.T) {
+                       testKeygen_compatibility(t, privateKey, publicKey,
+                               keygen_curve25519voi_ed25519, require.Equal, require.Equal)
+               })
```
```diff
+func keygen_curve25519voi_ed25519(secret [SEEDBYTES]byte) (ed25519.PrivateKey, ed25519.PublicKey) {
+       pubKey, privKey, _ := voied25519.GenerateKey(bytes.NewReader(secret[:]))
+       privateKey := make([]byte, voied25519.PrivateKeySize)
+       copy(privateKey, privKey[:])
+       publicKey := make([]byte, voied25519.PublicKeySize)
+       copy(publicKey, pubKey[:])
+       return privateKey, publicKey
+}
```
```diff
+func testKeypair_compatibility_curve25519voi_ed25519(
+       t *testing.T,
+       sk ed25519.PrivateKey, pk ed25519.PublicKey,
+       toPkTest func(t require.TestingT, expected interface{}, actual interface{}, msgAndArgs ...interface{}),
+       fromSkTest func(t require.TestingT, expected interface{}, actual interface{}, msgAndArgs ...interface{}),
+) {
+       pk1, sk1, _ := voied25519.GenerateKey(bytes.NewReader(secret[:]))
+
+       sk2 := ed25519.PrivateKey(make([]byte, ed25519.PrivateKeySize))
+       copy(sk2[:], sk1[:])
+       pk2 := sk2.Public().(ed25519.PublicKey)
+
+       toPkTest(t, pk1[:], pk2[:])
+
+       copy(sk1[:], sk[:])
+       pk1 = sk1.Public().(voied25519.PublicKey)
+
+       fromSkTest(t, pk1[:], pk2[:])
+}
```

And etc.

Reply:
Oh, I see. Thank you for your explanation.